src/config/kubefile.rs

```diff
@@ -104,6 +107,91 @@ pub struct ContextConf {
     pub user: String,
 }
 
+#[derive(Debug, Deserialize, Clone)]
+#[allow(non_snake_case)]
+pub struct Exec {
+    apiVersion: String,
+    pub args: Option<Vec<String>>,
+    pub command: Option<String>,
+    pub env: Option<Vec<Env>>,
+    pub token: RefCell<Option<String>>,
+    pub expiry: RefCell<Option<DateTime<Utc>>>,
+}
+
+#[derive(Debug, Deserialize, Clone)]
+pub struct Env {
+    name: String,
+    value: String,
+}
+
+#[derive(Serialize, Deserialize)]
+struct Spec {}
+
+#[derive(Serialize, Deserialize)]
+struct Status {
+    token: String,
+}
+
+#[derive(Serialize, Deserialize)]
+#[allow(non_snake_case)]
+struct AWSCredential {
+    kind: String,
+    apiVersion: String,
+    spec: Spec,
+    status: Status,
+}
+
+impl Exec {
+    // true if expiry is 10 minutes ago or more
+    fn check_dt<T: TimeZone>(&self, expiry: DateTime<T>) -> bool {
+        let etime = expiry.with_timezone(&Utc);
+        let now = Utc::now();
+        let diff = now.signed_duration_since(etime);
+        return diff.num_minutes() >= 10;
+    }
+
+    fn is_expired(&self) -> bool {
+        let expiry = self.expiry.borrow();
+        match *expiry {
+            Some(e) => {
+                self.check_dt(e)
+            }
+            None => {
+                true
+            }
+        }
+    }
+
+    fn generate_token(&self) -> String {
+        let mut filtered_env: HashMap<String, String> = HashMap::new();
```

Comment:
If we make the above change to have `self.env` be a HashMap, this shouldn't be needed. Below you can just do:
```rust
let output = Command::new(self.command.clone().unwrap())
            .args(self.args.clone().unwrap())
            .envs(&self.env)
            .output()
            .expect("failed to execute process");
```

Without testing myself I'm not sure you won't have to clone it, since `envs` want's an `IntoIter`, so you may need to do that.